tracee-ebpf/tracee/events.go

```diff
@@ -15,27 +15,51 @@ import (
 // NOTE: Integers want to be aligned in memory, so if changing the format of this struct
 // keep the 1-byte 'Argnum' as the final parameter before the padding (if padding is needed).
 type context struct {
-	Ts       uint64
-	Pid      uint32
-	Tid      uint32
-	Ppid     uint32
-	HostPid  uint32
-	HostTid  uint32
-	HostPpid uint32
-	Uid      uint32
-	MntID    uint32
-	PidID    uint32
-	Comm     [16]byte
-	UtsName  [16]byte
-	ContID   [16]byte
-	EventID  int32
-	Retval   int64
-	StackID  uint32
-	Argnum   uint8
-	_        [3]byte //padding
+	Ts             uint64
+	SMPProcessorId uint64
+	Pid            uint32
+	Tid            uint32
+	Ppid           uint32
+	HostPid        uint32
+	HostTid        uint32
+	HostPpid       uint32
+	Uid            uint32
+	MntID          uint32
+	PidID          uint32
+	Comm           [16]byte
+	UtsName        [16]byte
+	ContID         [16]byte
+	EventID        int32
+	Retval         int64
+	StackID        uint32
+	Argnum         uint8
+	_              [3]byte //padding
 }
 
-func (t *Tracee) processEvents(done <-chan struct{}) error {
+// handleEvents is a high-level function that starts all operations related to events processing
+func (t *Tracee) handleEvents(done <-chan struct{}) {
+	var closeChannels []chan struct{}
+	processEventsOutputChan := t.config.ChanEvents
+	if t.config.Sorting {
+		sorterInputChan := make(chan external.Event, 100)
```

Comment:
is the channel size arbitrary? how do we know is it a good amount? maybe 100 is not enough. Just thinking out loud. What do you think?

Reply:
You are right,
I am not sure what size is good for the channel.
Do you have any advice on how to determine the channel size?
In the end I hope the writing and reading speeds will be similar, and there shouldn't be any large peaks of differences except maybe when sending the events forward from the sorter (because part of the time it is blocking operation because all pointers and lists should be updated).